### PR TITLE
feat: handle connection requests errors (WPB-4418)

### PR DESCRIPTION
**Pull request by V-Gira**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4418" title="WPB-4418" target="_blank">BugWPB-4418</a>  [Web] handle connection request errors
  </td></table>
  <br />
 

## Description

We are not currently giving any feedback to the user if there's any error when sending / ignoring / canceling / accepting connection requests.

Other platforms have a generic error message (iOS), a toast for the different cases (android)

The decision (see ticket) was to use Androids error message in a modal

## Screenshots/Screencast (for UI changes)
Screenshot from 2023-09-11 15-37-19

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/i18n/en-US.json [hunk 1]
```diff
@@ -959,11 +959,14 @@
   "modalUserBlockHeadline": "Block {{user}}?",
   "modalUserBlockMessage": "{{user}} won’t be able to contact you or add you to group conversations.",
   "modalUserBlockedForLegalHold": "This user is blocked due to legal hold. [link]Learn more[/link]",
+  "modalUserCannotAcceptConnectionMessage": "Connection request could not be accepted",
   "modalUserCannotBeAddedHeadline": "Guests cannot be added",
-  "modalUserCannotConnectLegalHoldHeadline": "Cannot connect to this user",
```
Comment (V-Gira) [inline src/i18n/en-US.json:963]: The localization team requested to change this title for the no-longer-federating case.

We are using the new copy "Connecting not possible" as a title for every case

File: src/script/connection/ConnectionRepository.ts [hunk 2]
```diff
@@ -311,6 +318,39 @@ export class ConnectionRepository {
...
+              htmlMessage: t('modalUserCannotIgnoreConnectionMessage'),
+              title: t('modalUserCannotConnectHeadline'),
+            },
+          });
+        }
+        default: {
+          PrimaryModal.show(PrimaryModal.type.ACKNOWLEDGE, {
+            text: {
+              title: t('modalUserCannotConnectHeadline'),
+            },
```
Comment (V-Gira) [inline src/script/connection/ConnectionRepository.ts:350]: this shows a modal with no message, only the title "Connecting not possible" as a fallback for unhandled cases

Comment (aweiss-dev) [reply src/script/connection/ConnectionRepository.ts:350]: Doesn't it Look weird if the modal doesn't have a message?

**Review by aweiss-dev — APPROVED** [review]

**Review by atomrc — APPROVED** [review]

File: src/script/connection/ConnectionRepository.ts (hunk 2 above)
Comment (V-Gira) [reply src/script/connection/ConnectionRepository.ts:350]: looks fine to me, wdyt?
image

**Pull request merged**